homework/tests/homework.spec.js

```diff
@@ -0,0 +1,93 @@
+      await page.selectOption('#selectOperationDropdown', {label: 'Concatenate'});
+  
+      await expect(page.locator('#integerSelect')).toBeHidden();
+    });
+
+    test('Build 4: Integer only should be enabled when performing calculations', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: '4'});
+  
+      await expect(page.locator('#integerSelect')).toBeEnabled();
+    });
+
+    test('Build 5: Clear button should be enabled', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: '5'});
+  
+      await expect(page.locator('#clearButton')).toBeEnabled();
+    });
+
+    test('Build 6: Division by 0 should produce an error', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: '6'});
+      await page.locator('#number1Field').type('1');
+      await page.locator('#number2Field').type('0');
+      await page.selectOption('#selectOperationDropdown', {label: 'Divide'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#errorMsgField')).toBeVisible();
+    });
+
+    test('Build 7: Adding 0 and 1 twice should result in 1', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: '7'});
+      await page.locator('#number1Field').type('0');
+      await page.locator('#number2Field').type('1');
+      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
+      await page.locator('#calculateButton').click();
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('1');
+    });
+
+    test('Build 8: Typing a letter in Number Field 1 produces a corresponding error message', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: '8'});
+      await page.locator('#number1Field').type('a');
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#errorMsgField')).toHaveText('Number 1 is not a number');
+    });
+
+    test('Build 9: Number Field 2 should be visible', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: '9'});
+  
+      await expect(page.locator('#number2Field')).toBeVisible();
+    });
+
+    test('Build 9: Calculate button should be visible', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: '9'});
+  
+      await expect(page.locator('#calculateButton')).toBeVisible();
+    });
```

Comment:
These could go in a single test case, 2 expects could be used at the end of the test case